tooling/components/io.siddhi.distribution.editor.core/src/main/java/io/siddhi/distribution/editor/core/util/SourceEditorUtils.java

```diff
@@ -434,6 +439,246 @@ private static ProcessorMetaData generateProcessorMetaData(Class<?> processorCla
+    private static StringBuilder parameterDataTypeGeneration(String parameterName,
+                                                             HashMap<String, DataType[]> parameterMap) {
+
+        StringBuilder parameterDataType = new StringBuilder();
+        parameterDataType.append("<");
+        DataType[] parameterType = parameterMap.get(parameterName);
+        for (int i = 0; i < parameterType.length; i++) {
+            DataType dataType = parameterType[i];
+            if (i == parameterType.length - 1) {
+                parameterDataType.append(dataType);
+            } else {
+                parameterDataType.append(dataType).append("|");
+            }
+        }
+        parameterDataType.append(">");
+        return parameterDataType;
+    }
+
+    /*generate the syntax  for functions,windows,stream processor and aggregate
+     function*/
+    private static List<SyntaxMetaData> windowFunctionSyntaxGeneration(Extension extension,
+                                                                       HashMap<String, DataType[]> parameterMap,
+                                                                       List<SyntaxMetaData> syntaxList,
+                                                                       String finalClipBoardSyntax, String finalSyntax,
+                                                                       StringBuilder syntax,
+                                                                       StringBuilder clipBoardSyntax) {
+        //Add return syntax to the extension if it is a functions and aggregate functions
+        for (ReturnAttribute returnAttribute : extension.returnAttributes()) {
+            syntax.append("<");
+            for (int j = 0; j < (returnAttribute.type()).length; j++) {
+                if (j == returnAttribute.type().length - 1) {
+                    syntax.append(returnAttribute.type()[j]);
+                } else {
+                    syntax.append(returnAttribute.type()[j]).append("|");
+                }
+            }
+            syntax.append(">");
+        }
+        // Add name syntax to the extension
+        if (!extension.namespace().isEmpty()) {
+            syntax.append(extension.namespace()).append(":");
+            clipBoardSyntax.append(extension.namespace()).append(":");
+        }
+        syntax.append(extension.name());
+        clipBoardSyntax.append(extension.name());
+        //Add parameter and its data type syntax to extension based on the parameter overloads
+        if (extension.parameterOverloads().length > 0) {
+            StringBuilder parameterSyntax;
+            StringBuilder parameterSyntaxClipboard;
+            for (ParameterOverload parameterOverload : extension.parameterOverloads()) {
+                SyntaxMetaData syntaxMetaData = new SyntaxMetaData();
+                parameterSyntax = new StringBuilder();
+                parameterSyntaxClipboard = new StringBuilder();
+                String previousDataname;
```

Comment:
No need to initialise ALL variables at the top, if you are only using that temporarily, initialise only there, for instance this can be moved to the else block which follows